core/src/retransmit_stage.rs

```diff
@@ -23,27 +23,34 @@ use std::{
     sync::atomic::AtomicBool,
     sync::mpsc::channel,
     sync::mpsc::RecvTimeoutError,
+    sync::Mutex,
     sync::{Arc, RwLock},
     thread::{self, Builder, JoinHandle},
     time::Duration,
 };
 
-pub fn retransmit(
+fn retransmit(
     bank_forks: &Arc<RwLock<BankForks>>,
     leader_schedule_cache: &Arc<LeaderScheduleCache>,
     cluster_info: &Arc<RwLock<ClusterInfo>>,
-    r: &PacketReceiver,
+    r: &Arc<Mutex<PacketReceiver>>,
     sock: &UdpSocket,
+    id: u32,
 ) -> Result<()> {
     let timer = Duration::new(1, 0);
-    let packets = r.recv_timeout(timer)?;
+    let r_lock = r.lock().unwrap();
+    let packets = r_lock.recv_timeout(timer)?;
     let mut timer_start = Measure::start("retransmit");
     let mut total_packets = packets.packets.len();
     let mut packet_v = vec![packets];
-    while let Ok(nq) = r.try_recv() {
+    while let Ok(nq) = r_lock.try_recv() {
         total_packets += nq.packets.len();
         packet_v.push(nq);
+        if total_packets >= 100 {
```

Comment:
Maybe this should be a constant instead of a magic number. What was the reasoning behind 100? to off load work?